### PR TITLE
Make Add-Using project/metadata search run out of process

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Note: large size of the PR is because of the largely mechanical test updates.

**Comment by CyrusNajmabadi** [comment]
This updates add-using to be OOPified.  But the option is still set to false.

**Comment by CyrusNajmabadi** [comment]
This PR depends on https://github.com/dotnet/roslyn/pull/19854.

File: src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 1]
```diff
@@ -300,18 +301,33 @@ private static bool IsHostOrTestWorkspace(Project project)
...
+                        checksumCollection = checksumCollection ?? GetChecksumCollection(p, cancellationToken);
+                        var checksum = checksumCollection[i];
+
+                        result.Add((p.Id, peReference, checksum));
                     }
                 }
             }
 
-            return result.ToImmutableAndFree();
+            return result.ToImmutableArray();
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: Should still be `ToImmutableAndFree()`.

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public CSharpWinForms(VisualStudioInstanceFactory instanceFactory)
         {
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.WinForms)]
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/19858"), Trait(Traits.Feature, Traits.Features.WinForms)]
```
Comment (Pilchie) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs]: @jasonmalinowski has been trying to get more logging stuff here.

File: src/Features/Core/Portable/AddImport/AddImportSymbolTreeInfoService.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            var result = ArrayBuilder<SymbolAndProjectId>.GetInstance();
+
+            foreach (var value in array)
+            {
+                // Don't force getting the symbol on this end.  We don't want to perform
+                // an expensive computation locally as that may slow down Add-Import greatly.
+                var symbolAndProjectId = await value.TryRehydrateAsync(
+                    solution, forceCompilation: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                result.AddIfNotNull(symbolAndProjectId);
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/AddImportSymbolTreeInfoService.cs]: I don't understand.  Does this mean we're just dropping results on the floor if this process doesn't have them?

**Comment by CyrusNajmabadi** [comment]
Talked to pilchie.  Let's go another route here.

**Pull request closed without merging**